### ISSUE TITLE
backtick quotes in TypeScript types

### ISSUE THREAD

**Issue by huseeiin** [eslint-stylistic/eslint-stylistic#473]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

```ts
// "@stylistic/quotes": [`error`, `backtick`]
// this rule should ignore types as this is a syntax error:
type A = import(`hi`);
```

### Reproduction

production is above ^

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
